lib/request-execution.js

```diff
@@ -149,7 +149,10 @@ class RequestExecution {
     };
     const self = this;
     function onRequestError() {
-      return self._parent.retryPolicy.onRequestError(operationInfo, self._request.consistency, err);
+      if(operationInfo.options.isIdempotent) {
```

Comment:
I was going to suggest to short circuit above, at the beginning of `_getDecision()` method, but I see that when `err.isSocketError && err.requestNotWritten` it can be safely retried on next host... 👍